src/rules.jl

```diff
@@ -108,6 +108,28 @@ See also: [`accumulate`](@ref), [`accumulate!`](@ref), [`AbstractRule`](@ref)
 """
 store!(Δ, rule::AbstractRule, args...) = materialize!(Δ, broadcastable(rule(args...)))
 
+# Special purpose updating for operations which can be done in-place
```

Comment:
Are these just helper functions for `accumulate!`?

Reply:
Yes, I didn't make them methods of `accumulate!` because they don't share the same expected method signature form, i.e. `accumulate!(value, rule, args)`; this is more free-form.

Reply:
Fair enough. Could you please extend the comment to this effect? i.e. the reason for `_update!` existing / its intended use?